### PR TITLE
backport https://github.com/xapi-project/xen-api/pull/4646 to yangtze

### PR DESCRIPTION
**Pull request by liulinC**
This is backport of  https://github.com/xapi-project/xen-api/pull/4646, target to Yangtze

### CODE REVIEW

File: scripts/plugins/extauth-hook-AD.py [hunk 1]
```diff
@@ -205,18 +244,19 @@ def is_special_line(self, line):
 
     def _load_values(self):
```
Comment (lindig) [inline scripts/plugins/extauth-hook-AD.py:245]: I understand this is a packport but I notice a deeply nested logic that I don't trust to be correct. There are several special cases (empty lines, kv != 2) that are mostly unmotivated. Given that this code handles part of authentication I would like to see more clarity.

**Review by lindig — APPROVED** [review]

**Review by psafont — CHANGES_REQUESTED** [review]
Please add the hash of the original commits in the commit message of every commit, this makes it easier to see that the backport is accurate.

**Comment by psafont** [comment]
The first commit is not a cherry-pick, it has additonal changes not present in the original commit. Is there a reason for this? If there are more commit backported please add each one as a cherry-pick.

The strategy with backport is to only import commits that are already on `master`, plus additional commits when some adaptations are needed to make it work on the branch

**Comment by liulinC** [comment]
> Please add the hash of the original commits in the commit message of every commit, this makes it easier to see that the backport is accurate.

Just  updated, p

> The first commit is not a cherry-pick, it has additonal changes not present in the original commit. Is there a reason for this? If there are more commit backported please add each one as a cherry-pick.
> 
> The strategy with backport is to only import commits that are already on `master`, plus additional commits when some adaptations are needed to make it work on the branch

My autopep vim plugin updated it for me, I have just fixup, please help to take a look, thanks.

**Review by psafont — APPROVED** [review]
On the first commit a few more changes were added to match the final output of the patch.

Looks like a faithful backport to me.

**Comment by liulinC** [comment]
> On the first commit a few more changes were added to match the final output of the patch.
> 
> Looks like a faithful backport to me.

The original commit do include a little pep8 fixup, see https://github.com/xapi-project/xen-api/pull/4646/commits/93dc7707fd4fa4de55a285ca9a179c3c1124ede0

That is caused by refine a little pep8 during the issue, then I put the remaining to a single commit.
I should squash the commits in master, but since it got merged, let's keep consistency here.

**Pull request merged**